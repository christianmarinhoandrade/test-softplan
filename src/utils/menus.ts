export const getMenu = (
) => {
  return [
    {
      title: 'teste',
      path: '/teste',
      name: 'agendamento'
    }
  ]
}
