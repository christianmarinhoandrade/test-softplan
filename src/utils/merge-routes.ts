import { RouteProps } from 'react-router-dom'
const routesContext = require.context('../pages', true, /.routes./)

export type RouteMergeProps = {
  name?: string
  path?: string
} & RouteProps

const routes: RouteMergeProps[] = []

routesContext.keys().forEach((route) => {
  routes.push(...routesContext(route).default)
})

export default routes
