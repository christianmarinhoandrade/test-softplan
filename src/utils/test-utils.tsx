import { ReactElement } from 'react'
import { render } from '@testing-library/react'

import { Provider as ProviderRedux } from 'react-redux'
import store from 'store'

import ThemeProvider from 'providers/theme'

const customRender = (ui: ReactElement) =>
  render(
    <ProviderRedux store={store}>
      <ThemeProvider>{ui}</ThemeProvider>
    </ProviderRedux>
  )

export * from '@testing-library/react'
export { customRender as render }
