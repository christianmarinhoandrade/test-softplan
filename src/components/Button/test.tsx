import { screen } from '@testing-library/react'
import { render } from 'utils/test-utils'

import { CalendarEdit } from '@styled-icons/boxicons-regular'

import { Button } from '.'

describe('<Button />', () => {
  it('should render by default', () => {
    const { container } = render(<Button>Buy now</Button>)

    expect(screen.getByRole('button', { name: /Buy now/i })).toHaveStyle({
      width: '100%'
    })

    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render with width and height', () => {
    render(
      <Button width="40px" height="30px">
        Buy now
      </Button>
    )

    expect(screen.getByRole('button', { name: /Buy now/i })).toHaveStyle({
      width: '40px',
      height: '30px'
    })
  })

  it('should render an icon version', () => {
    render(<Button icon={<CalendarEdit data-testid="icon" />}>Buy now</Button>)

    expect(screen.getByText(/buy now/i)).toBeInTheDocument()
    expect(screen.getByTestId('icon')).toBeInTheDocument()
  })

  it('should render with background color', () => {
    render(<Button backgroundColor={'#e64a0d'}>Buy now</Button>)

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyleRule(
      'background-color',
      '#e64a0d'
    )
  })

  it('should render a disabled Button', () => {
    render(<Button disabled={true}>Buy now</Button>)

    expect(screen.getByRole('button', { name: /buy now/i })).toHaveStyleRule(
      'cursor',
      'not-allowed',
      {
        modifier: ':disabled'
      }
    )
  })
})
