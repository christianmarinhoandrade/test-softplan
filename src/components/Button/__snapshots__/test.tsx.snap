// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should render by default 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 1rem;
  font-family: Open Sans;
  cursor: pointer;
  border: none;
  border-radius: 0.2rem;
  background-color: #3b97de;
  color: #001965;
  width: 100%;
}

<button
  class="c0"
>
  <span>
    Buy now
  </span>
</button>
`;
